### PR TITLE
372068: Automate Role Assignment

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: *

[AB#372068](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/372068)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*
Depends on PR: https://github.com/DEFRA/adp-pipeline-common/pull/182

# Testing
*Any relevant testing information and pipeline runs.*
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=549467&view=results
image

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: .azuredevops/deploy-platform-core-iam.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      ref: main
+      
+variables:
+  - name: IsAll
+    value: ${{ eq(parameters.deployResources,'All') }}
+  - name: IsClusterCustomResourcesReaderRole
+    value: ${{ or(eq(variables.IsAll,true), eq(parameters.deployResources,'Cluster Custom Resources Reader Role')) }}    
+
+extends:
+  template: /pipelines/common-infra-deploy.yaml@ADPPipelineCommon
```
Comment (DaRoszko) [inline .azuredevops/deploy-platform-core-iam.yaml:47]: can't this inherit ADP common? the abstraction layer for envs already exists?

Comment (rajeshkaremane) [reply .azuredevops/deploy-platform-core-iam.yaml:47]: Removed env section.

File: .azuredevops/deploy-platform-core-iam.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+name: 1.0.$(BuildID)-${{ parameters.deployResources }}
```
Comment (ajaysoni-cap) [inline .azuredevops/deploy-platform-core-iam.yaml:1]: Why is this a separate pipeline?

Comment (ajaysoni-cap) [reply .azuredevops/deploy-platform-core-iam.yaml:1]: This is fine as discussed in the meeting.

Comment (rajeshkaremane) [reply .azuredevops/deploy-platform-core-iam.yaml:1]: As discussed it is due to the dependency of app reg being created before we assign role

**Review by kbabigumira — APPROVED** [review]

**Review by ajaysoni-cap — APPROVED** [review]

**Pull request merged**